dataactbroker/README.md

```diff
@@ -773,6 +773,153 @@ Example output if there are no files available:
 }
 ```
 
+# File Generation Routes
+
+## Generate Files
+**Route:** `/generate_file`
+
+**Method:** `POST`
+
+This route sends a request to the backend to utilize the relevant external APIs and generate the relevant file for the metadata that is submitted.
+
+**Deprecation Notice:** This route replaces `/generate_d1_file` and `/generate_d2_file`.
+
+### Body (JSON)
+
+```
+{
+    "submission_id": 123,
+    "file_type": "D1"
+    "start": "01/01/2016",
+    "end": "03/31/2016"
+}
+```
+
+### Body Description
+
+* `submission_id` - **required** - an integer representing the ID of the current submission
+* `file_type` - **required** - a string indicating the file type to generate. Allowable values are:
+	* `D1` - generate a D1 file
+	* `D2` - generate a D2 file
+	* `E` - generate a E file
+	* `F` - generate a F file
+* `start` - **required for D1/D2 only** - the start date of the requested period, in `MM/DD/YYYY` string format
```

Comment:
Changed "period" to "date range", let me know if there's another term that would be clearer